### PR TITLE
Documentation updates

### LINKED ISSUE #238: Canonical way to run inference on a trained model?
Let's say I have a trained `NeuralNet` model and one sample of data, and I want to infer network's output. Ideally, I'd like to have a method that:

1. Accepts non-batched data.
2. Handles cpu/gpu transfers transparently
2a. Data are automatically transferred to gpu if network is on gpu
2b. Output is on the same device as input (or possible to select explicitly)
3. In case the network has several output values, gives me all of them.
4. Does not compute gradients.

There are several methods with names that sound suitable for the task (`infer`, `forward`, `predict`), however none of them fits all the requirements I outlined. Despite having the least "suitable" name, `forward` actually satisfies most of the requirements, though I still need to explicitly unsqueeze/squeeze my data to use it.

Is there anything I missed?

### PR DESCRIPTION
**Pull request by taketwo**
This PR contains a bunch of minor documentation fixes, see individual commit messages. Closes #238.

### CODE REVIEW

File: docs/user/neuralnet.rst [hunk 1]
```diff
@@ -326,20 +327,20 @@ predict(X) and predict_proba(X)
...
 
-If casting the :func:`~torch.nn.Module.forward`\-output to ``numpy``
-is impossible, you will get an error. In that case, you should
-consider returning a PyTorch :class:`~torch.Tensor` from your
-:func:`~torch.nn.Module.forward` method, as this tensor can be
-converted to a ``numpy`` array. Alternatively, consider using the
-:func:`~skorch.net.NeuralNet.forward_iter` method to generate outputs
-from the ``module``, or directly call ``net.module_(X)``.
+If the :func:`~torch.nn.Module.forward`\-output can not be meaningfuly
+casted to a :class:`numpy.ndarray`, or if you need access to all
```
Comment (BenjaminBossan) [inline docs/user/neuralnet.rst]: ~casted~ cast

**Comment by taketwo** [comment]
Frankly, this is a mistake, not a typo :) Will update this PR on Monday.

**Comment by taketwo** [comment]
Done.

File: docs/user/neuralnet.rst [hunk 2]
```diff
@@ -326,20 +327,20 @@ predict(X) and predict_proba(X)
...
 only the first output is used, the rest is discarded.
 
-If casting the :func:`~torch.nn.Module.forward`\-output to ``numpy``
-is impossible, you will get an error. In that case, you should
-consider returning a PyTorch :class:`~torch.Tensor` from your
-:func:`~torch.nn.Module.forward` method, as this tensor can be
-converted to a ``numpy`` array. Alternatively, consider using the
-:func:`~skorch.net.NeuralNet.forward_iter` method to generate outputs
-from the ``module``, or directly call ``net.module_(X)``.
+If the :func:`~torch.nn.Module.forward`\-output can not be meaningfuly
```
Comment (benjamin-work) [inline docs/user/neuralnet.rst]: Sorry, I noticed the typo in "meaningfuly" just now. But I wonder whether this word is needed here at all. If the passage just reads "... cannot be cast to ...", the meaning should be clear.

Comment (taketwo) [reply docs/user/neuralnet.rst]: Agree, fixed.

**Comment by benjamin-work** [comment]
Great, thanks.
